verticapy/stats/tools.py

```diff
@@ -1424,3 +1425,124 @@ def variance_inflation_factor(vdf: vDataFrame, X: list, X_idx: int = None):
         raise ParameterError(
             f"Wrong type for Parameter X_idx.\nExpected integer, found {type(X_idx)}."
         )
+
+# ---#
+def is_dir_path(adj_list: dict, X:str, Y:str):
+    """
+---------------------------------------------------------------------------
+Determines if there is a directed path from X -> Y given an adjacency list.
+
+Parameters
+----------
+adj_list: dict
+    Adjacency List where the values are the other keys
+X: str
+    Starting node
+Y: str
+    Ending Node
+
+Returns
+-------
+Boolean 
+True if there exists a path
+    """
+    if X not in adj_list.keys():
+        raise ParameterError(f"{X} not in Adjacency List")
+    if Y not in adj_list.keys():
+        raise ParameterError(f"{Y} not in Adjacency List")
+
+    visited = [X]
+    queue = [X]
+
+    while(queue):
+        m = queue.pop(0)
+        for neighbor in adj_list[m]:
+            if m not in adj_list[neighbor]: 
```

Comment:
I don't understand this if-statement!
Let's your graph as only X and Y nodes with a cycle:
`adj_list = {{X, {Y}}, {Y, {X}}}`
There is a path from X to Y and a path from Y to X, but the implemented function will deny existence of these paths. Do I miss something?

Reply:
This is because there isn't a directed path between these two points. There would be an undirected edge between X<->Y and this function is only looking for directed connections.